Libraries/Modal/RCTModalHostViewNativeComponent.js

```diff
@@ -42,6 +42,12 @@ type NativeProps = $ReadOnly<{|
     'fullScreen',
   >,
 
+  /**
+   * The `detent` determines the height for modal,
+   * based on the detent value.
+   */
+  detent?: WithDefault<'large' | 'medium' | 'mediumResizable', 'large'>,
```

Comment:
It might be to late to do this now - but should these props be `detents: Array<'large' | 'medium'>` and `resizable`? Reason being iOS 16+ supports custom-sized detents, and they could be easily added in a future PR if we get this API correct now

Reply:
Yes, you are probably right. The iOS [Detents](https://developer.apple.com/documentation/uikit/uisheetpresentationcontroller/3801903-detents) property is also an `Array` so it makes sense to mimic that in JS.

I'm only worried that we are not providing any mechanism to create a custom sized detent, right now, but we can probably live with it for the moment.

@hurali97 could you take care to update the PR as suggested by @janicduplessis, please? 🙏

Reply:
Sure fellas, I can incorporate to these changes 🚀

Reply:
Guys, we have a situation:

We check for a single detent property which is set from RN and we apply a switch statement to handle different cases. If we go with arrays for detents, we may lose the control on the prop, as it will be hard to apply guards on an array prop.

 One solution could be to allow detents array and set them on VC as is, but I don't fancy this one it's more prune to error. 

Second one is to allow the single detent prop so we have the control on it. To support custom sized detent, we can add it to detents enum and then handle it. I am in favour of this one.

Need your reviews on this thing as well 👀

```.mm
- (void)handleDetent:(ModalHostViewDetent)detent
{
  if (@available(iOS 15.0, *)) {
    if (!self.viewController.sheetPresentationController) {
      return;
    }

    NSArray<UISheetPresentationControllerDetent *> *detents = @[UISheetPresentationControllerDetent.largeDetent];
    switch (detent) {
      case ModalHostViewDetent::Large:
            break;
      case ModalHostViewDetent::Medium:
        detents = @[UISheetPresentationControllerDetent.mediumDetent];
        break;
      case ModalHostViewDetent::MediumResizable:
        self.viewController.sheetPresentationController.prefersGrabberVisible = true;
        self.viewController.sheetPresentationController.largestUndimmedDetentIdentifier =  UISheetPresentationControllerDetentIdentifierMedium;
        detents = @[UISheetPresentationControllerDetent.mediumDetent, UISheetPresentationControllerDetent.largeDetent];
        break;
    }
    self.viewController.sheetPresentationController.detents = detents;
  }
}
```

Reply:
Delete your enum definition, store the prop as type `id`, and then do the conversion in a guard when presenting the modal

Reply:
Just to be on the same page, are we still referring to detent as an array ? 

If yes, I want to discuss to apply some custom code for `mediumResizable` we will need to iterate thru the array. Such is the case with custom sized detent.

Reply:
You'd have one array for `detents`, and one boolean prop `resizable`. So your `medium-resiable` would be expressed like

```js
<Modal detents={['medium', 'large']} resizable={true} />
```

Reply:
I am currently working on these changes but since Xcode 14 updates, rn-tester isn't building even after pulling in the latest changes.

Reply:
Thinking more about this - you can probably just assume `resizable` if they pass in more than one detent. The only use-case is for imperatively handling detents - but I don't see anyone that cares enough about that to actually implement it.

```objc
_viewController.sheetPresentationController.prefersGrabberVisible = [detents count] > 1;
_viewController.sheetPresentationController.largestUndimmedDetentIdentifier =  UISheetPresentationControllerDetentIdentifierMedium;
```

I don't have a laptop with RN set up on it - and won't for the rest of the week - but if you need a hand with this, just shout and I can help you once I get hold of it again (and get some time too!)

Reply:
Hey @jacobp100, thanks for the discussion. I tend to agree with you and also I was thinking to remove `_viewController.sheetPresentationController.largestUndimmedDetentIdentifier` as we probably not want to let the touch passed down to the undimmed area.


Apart from that, I have updated the implementation to support the arrays with custom height(_it's just a placeholder 80 for now in RN_):

```
      <Modal
        detents={['80', 'medium', 'large']}
     />
```


Now the thing is, if we go back to iOS15 we had medium and large detents, which works very well, like we can swipe down to dismiss the modal and all, because the `onRequestClose` is fired so we can implement `onDismiss` on RN.

Notice that I added 80 as a custom height and it works very well but it doesn't fire `presentationControllerDidAttemptToDismiss` when swiped down. I am currently finding it hard to understand how to handle this behaviour. If I remove this custom sized detent, all is working as expected means I am able to dismiss the modal.


Below is the code that introduces that custom sized detent, that's currently a rough code:

```.mm
        for (NSString *detent in _detents) {
         if ([detent isEqualToString:@"medium"]) {
            [detents addObject:UISheetPresentationControllerDetent.mediumDetent];
         } else if ([detent isEqualToString:@"large"]) {
              [detents addObject:UISheetPresentationControllerDetent.largeDetent];
         } else {
             /**
                Check if item type is number??
              */
             if (@available(iOS 16.0, *)) {
                 UISheetPresentationControllerDetent *customDetent = [UISheetPresentationControllerDetent customDetentWithIdentifier:nil resolver:^CGFloat(id<UISheetPresentationControllerDetentResolutionContext>  _Nonnull context) {
                           return 80;
                       }];
                 [detents addObject:customDetent];
             }
         }
        }

```
https://user-images.githubusercontent.com/47336142/198103031-214f90aa-d6fa-444d-985a-e012f4721153.mp4

Reply:
I was playing a bit with heights and setting 120 or above dismisses the sheet when swiped down. I think there's some sort of threshold internally? I might be wrong, need to look more 👀